File: Aug/Day26/maximum-area-of-longest-diagonal-rectangle.py
#  Time: O(N)  ::  Space: O(1)
class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        longestDiagonal = 0
        areaOfLongestDiagonal = 0

        for length, width in dimensions:
            diagonal = math.sqrt((length*length)+(width*width))
            area = length*width
            if diagonal > longestDiagonal:
                areaOfLongestDiagonal = area
                longestDiagonal = diagonal 
            elif diagonal == longestDiagonal:
                areaOfLongestDiagonal = max(area, areaOfLongestDiagonal)
        return areaOfLongestDiagonal
        

File: Aug/Day26/maximum-area-of-longest-diagonal-rectangle.js
//  Time: O(N)  ::  Space: O(1)
class Solution {
    areaOfMaxDiagonal(dimensions) {
        let longestDiagonal = 0;
        let areaOfLongestDiagonal = 0;

        for (let [length, width] of dimensions) {
            let diagonal = Math.sqrt((length * length) + (width * width));
            let area = length * width;

            if (diagonal > longestDiagonal) {
                longestDiagonal = diagonal;
                areaOfLongestDiagonal = area;
            } else if (diagonal === longestDiagonal) {
                areaOfLongestDiagonal = Math.max(areaOfLongestDiagonal, area);
            }
        }
        return areaOfLongestDiagonal;
    }
}
